### PR TITLE
[JENKINS-51956] upgrade to plugins parent 3.15

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <olamy@apache.org>

# Description

See [JENKINS-51956](https://issues.jenkins-ci.org/browse/JENKINS-51956).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by vivek** [comment]
@olamy build is failing.

**Comment by olamy** [comment]
working on it

File: pom.xml [hunk 1]
```diff
@@ -854,6 +854,15 @@
   <build>
       <pluginManagement>
           <plugins>
+              <plugin>
+                  <groupId>org.kohsuke</groupId>
+                  <artifactId>access-modifier-checker</artifactId>
```
Comment (vivek) [inline pom.xml]: @olamy probably ok. @jglick does it introduce/mask problems it shouldn't?

Comment (jglick) [reply pom.xml]: Yes.

Comment (vivek) [reply pom.xml]: @jglick ok, so I take it as not a good idea to introduce. @olamy what its solving? What kind of error we are trying to fix with this?

Comment (olamy) [reply pom.xml]: BlueUrlTokenizer (module blueocean-commons) is marked with @Restricted(NoExternalUse.class).
This is used by many classes outside this module (PipelineActivityStatePreloader from blueocean-pipeline-api-impl etc..)
I guess it's an internal use :-)
So here few solutions:
1. skip access-modifier-checker
2. rewrite/remove BlueUrlTokenizer
3. fix access-modifier-checker plugin to accept usage with @Restricted(NoExternalUse.class) from modules within the same project (reactors)

I would say 3 (I need to have a look how long will be the fix)
Let me know what you prefer?

Comment (olamy) [reply pom.xml]: I have implemented/pushed solution 4.

File: pom.xml [hunk 2]
```diff
@@ -42,6 +42,9 @@
     <jenkins-test-harness.version>2.38</jenkins-test-harness.version>
     <scm-api.version>2.2.6</scm-api.version>
     <git.version>3.8.0</git.version>
+    <!-- upgrading to parent 3.15 introduce a new version which is VERY MORE MORE restrictive so skip it
+    as this need a lot of changes.... -->
+    <access-modifier-checker.failOnError>false</access-modifier-checker.failOnError>
```
Comment (vivek) [inline pom.xml]: @olamy This should always be true, no? We do not want to disable this check.

Comment (olamy) [reply pom.xml]: ohh gloupss I changed the code but not this one.
Fixed!

Comment (olamy) [reply pom.xml]: Well we have more problems now....
jenkins.branch.OrganizationFolder from branch-api is marked as @Restricted({NoExternalUse.class}) and we use it.
As it's external, I cannot move it.

**Pull request merged**